### ISSUE TITLE
LibriSpeech Wav2Vec 2.0 recipe: why do we need detach()?

### ISSUE THREAD

**Issue by EugKar** [speechbrain/speechbrain#1077]
Hello. Looking through your Librispeech Wav2Vec2 recipe, I found a line https://github.com/speechbrain/speechbrain/blob/39840decb0ce2b4f2f165e5df318748d2f4df019/recipes/LibriSpeech/ASR/ctc/train_with_wav2vec.py#L55
If I'm understanding correctly, gradients are not propagated to Wav2Vec module, regardless of the settings in the [.yaml](https://github.com/speechbrain/speechbrain/blob/develop/recipes/LibriSpeech/ASR/ctc/hparams/train_with_wav2vec.yaml) file. Whereas in the .yaml file there is https://github.com/speechbrain/speechbrain/blob/39840decb0ce2b4f2f165e5df318748d2f4df019/recipes/LibriSpeech/ASR/ctc/hparams/train_with_wav2vec.yaml#L62 which is then passed to the `HuggingFaceWav2Vec2` constructor. There is also a separate optimizer for Wav2Vec. So, it looks like there is a bug in the training code or in the .yaml file.
